ogr/ogrsf_frmts/xodr/ogrxodrdrivercore.cpp

```diff
@@ -0,0 +1,89 @@
+/******************************************************************************
+ *
+ * Project:  OpenGIS Simple Features for OpenDRIVE
+ * Purpose:  Implementation of OGRXODRDriverCore.
+ * Author:   Michael Scholz, German Aerospace Center (DLR)
+ *           Gülsen Bardak, German Aerospace Center (DLR)        
+ *
+ ******************************************************************************
+ * Copyright 2024 German Aerospace Center (DLR), Institute of Transportation Systems
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "ogrsf_frmts.h"
+#include "ogrxodrdrivercore.h"
+
+/************************************************************************/
+/*                    OGRXODRDriverIdentify()                            */
+/************************************************************************/
+
+int OGRXODRDriverIdentify(GDALOpenInfo *poOpenInfo)
+{
+    CPLString extension = CPLString(CPLGetExtension(poOpenInfo->pszFilename));
+    extension.tolower();
+    return EQUAL(extension, "xodr");
```

Comment:
EQUAL() is case insensitive

```suggestion
    return EQUAL(CPLGetExtension(poOpenInfo->pszFilename), "xodr");
```